### PR TITLE
Delete .Next projects

### PR DESCRIPTION
**Pull request by jasonmalinowski**
The .Next projects existed to support running Roslyn on both Dev14 and Dev15. We no longer need this support, it was somewhat broken already, and it is in the way of the next refactoring we need to do. Thus, we merge all the projects back together.

VisualStudioSetup.Next still *does* exist, but it is now empty. In order for F5 to work, we need to ensure we override the VSIX assets that are installed with VS, and thus we need an empty VSIX to do that with. We're no longer deploying any contents with it though, so it's effectively a no-op.

### CODE REVIEW

File: src/EditorFeatures/Core/Extensibility/Composition/VisualStudioVersionMetadata.cs [hunk 1]
```diff
@@ -1,16 +0,0 @@
-﻿using System;
-using System.Collections.Generic;
-using Roslyn.Utilities;
-
-namespace Microsoft.CodeAnalysis.Editor.Extensibility.Composition
-{
-    internal sealed class VisualStudioVersionMetadata
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Extensibility/Composition/VisualStudioVersionMetadata.cs]: Shouldn't we keep this.  Just for when we have v15 and v16 as .next?

File: src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Presentation/RoslynCompletionSet.cs [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.CodeAnalysis.Text.Shared.Extensions;
+using Microsoft.VisualStudio.Language.Intellisense;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using VSCompletion = Microsoft.VisualStudio.Language.Intellisense.Completion;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense.Completion.Presentation
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Presentation/RoslynCompletionSet.cs]: why so green?

File: src/EditorFeatures/Next/IntelliSense/Completion/Presentation/Roslyn15CompletionSet.cs [hunk 3]
```diff
@@ -1,104 +0,0 @@
...
-    /// <summary>
-    /// See comment on <see cref="VisualStudio15CompletionSet"/> for an explanation of how
-    /// these types fit together and where code should go in them.
-    ///
-    /// This class is where all code that would normally be in our derived type of 
-    /// <see cref="CompletionSet2"/> should go.  <see cref="VisualStudio15CompletionSet"/> 
-    /// should then just forward all calls to us and we can share all our Dev14 logic with 
-    /// <see cref="Roslyn14CompletionSet"/>.
-    /// </summary>
-    internal class Roslyn15CompletionSet : Roslyn14CompletionSet
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Next/IntelliSense/Completion/Presentation/Roslyn15CompletionSet.cs]: to make diff cleaner, these could be moved out of .next, but made partial.  so they look more like moves, and not rewrites.

File: src/VisualStudio/Core/Def/Implementation/DebuggerIntelliSense/DebuggerIntellisenseWorkspace.cs [hunk 4]
```diff
@@ -5,7 +5,7 @@
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.DebuggerIntelliSense
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/Implementation/DebuggerIntelliSense/DebuggerIntellisenseWorkspace.cs]: i would personally prefer we move to Mirosoft.CodeAnalysis.VisualStudio

**Comment by CyrusNajmabadi** [comment]
I really hate the Microsoft.CodeAnalysis.Workspace stuff.

File: src/EditorFeatures/Core/Extensibility/Composition/VisualStudioVersionMetadata.cs (hunk 1 above)
Comment (jasonmalinowski) [inline src/EditorFeatures/Core/Extensibility/Composition/VisualStudioVersionMetadata.cs]: @CyrusNajmabadi We can bring it back if needed. And TBH, after doing this exercise, I'm not sure I'd say we should try this approach again. With the VS Installer now allowing side-by-side with much less drama, just having separate branches might be a lot easier.

File: src/EditorFeatures/Next/IntelliSense/Completion/Presentation/Roslyn15CompletionSet.cs (hunk 3 above)
Comment (jasonmalinowski) [inline src/EditorFeatures/Next/IntelliSense/Completion/Presentation/Roslyn15CompletionSet.cs]: @CyrusNajmabadi Will that help? I think the bigger problem here is me flattening the hierarchy, so Git lost track of what was going on.

File: src/VisualStudio/Core/Def/Implementation/DebuggerIntelliSense/DebuggerIntellisenseWorkspace.cs (hunk 4 above)
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/Implementation/DebuggerIntelliSense/DebuggerIntellisenseWorkspace.cs]: @CyrusNajmabadi That'd cause a lot of other downstream problems (some of this is a public API after all.) Am I missing something here?

Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/Implementation/DebuggerIntelliSense/DebuggerIntellisenseWorkspace.cs]: Oh, it's public?  Poop.  Ok.  LGTM then

File: src/EditorFeatures/Next/IntelliSense/Completion/Presentation/Roslyn15CompletionSet.cs (hunk 3 above)
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Next/IntelliSense/Completion/Presentation/Roslyn15CompletionSet.cs]: i think you could flatten, but  by using partial classes.

File: src/EditorFeatures/Core/EditorFeatures.csproj [hunk 5]
```diff
@@ -53,14 +53,15 @@
     <PackageReference Include="Microsoft.VisualStudio.Threading" Version="$(MicrosoftVisualStudioThreadingVersion)" />
     <PackageReference Include="Microsoft.CodeAnalysis.Elfie" Version="$(MicrosoftCodeAnalysisElfieVersion)" />
     <PackageReference Include="Microsoft.VisualStudio.RemoteControl" Version="$(MicrosoftVisualStudioRemoteControlVersion)" />
+    <PackageReference Include="VS.ExternalAPIs.CodingConventions" Version="$(VSExternalAPIsCodingConventionsVersion)" />
+    <!-- The VS.ExternalAPIs.CodingConventions package isn't a "true" NuGet package and doesn't have the right lib folders. Manually reference the binary. -->
+    <Reference Include="$(NuGetPackageRoot)\VS.ExternalAPIs.CodingConventions\$(VSExternalAPIsCodingConventionsVersion)\Microsoft.VisualStudio.CodingConventions.dll" />
```
Comment (tmat) [inline src/EditorFeatures/Core/EditorFeatures.csproj]: >VS.ExternalAPIs.CodingConventions [](start = 44, length = 33)

What is this good for?

Comment (jasonmalinowski) [reply src/EditorFeatures/Core/EditorFeatures.csproj]: This is the VS .editorconfig library.

**Comment by jasonmalinowski** [comment]
@dotnet-bot retest windows_debug_unit64_prtest please.

**Pull request merged**